### PR TITLE
ros2: Parse camera objects parameters in camera node

### PR DESCRIPTION
**Pull request by Erblinium**

### CODE REVIEW

**Comment by benthie** [comment]
Thank you for the fix. It looks good to me.

It would be nice if you could add "ros2: "at the beginning of your commit message, because the changes are actually only related to the ROS2 part in which I forgot to add the mentioned parameters.

As soon as the code-check passes we can merge.

**Review by benthie — APPROVED** [review]

File: ensenso_camera2/launch/stereo_node.launch.py [hunk 1]
```diff
@@ -16,6 +16,9 @@ def launch_setup(context, *args, **kwargs):
     link_frame = ensenso.fix_empty_string(LaunchConfiguration("link_frame").perform(context))
     robot_frame = ensenso.fix_empty_string(LaunchConfiguration("robot_frame").perform(context))
     wrist_frame = ensenso.fix_empty_string(LaunchConfiguration("wrist_frame").perform(context))
+    objects_file = ensenso.fix_empty_string(LaunchConfiguration("objects_file").perform(context))
+    objects_frame = ensenso.fix_empty_string(LaunchConfiguration("objects_frame").perform(context))
+    visualization_marker_topic = ensenso.fix_empty_string(LaunchConfiguration("visualization_marker_topic").perform(context))
```
Comment (benthie) [inline ensenso_camera2/launch/stereo_node.launch.py:21]:
```
    visualization_marker_topic = ensenso.fix_empty_string(
        LaunchConfiguration("visualization_marker_topic").perform(context)
    )
```

File: ensenso_camera2/launch/stereo_node.launch.py [hunk 2]
```diff
@@ -35,6 +38,10 @@ def launch_setup(context, *args, **kwargs):
                 "wrist_frame": wrist_frame,
                 "tcp_port": LaunchConfiguration("tcp_port"),
                 "wait_for_camera": LaunchConfiguration("wait_for_camera"),
+                "objects_file": objects_file,
+                "objects_frame": objects_frame,
+                "visualization_marker_topic": visualization_marker_topic,
+                "visualization_marker_rate": LaunchConfiguration("visualization_marker_rate")
```
Comment (benthie) [inline ensenso_camera2/launch/stereo_node.launch.py:44]:
```
"visualization_marker_rate": LaunchConfiguration("visualization_marker_rate"),
```

File: ensenso_camera2/launch/stereo_node.launch.py [hunk 3]
```diff
@@ -55,6 +62,10 @@ def generate_launch_description():
     wrist_frame_launch_arg = DeclareLaunchArgument("wrist_frame", default_value=ensenso.EMPTY_STRING)
     tcp_port_launch_arg = DeclareLaunchArgument("tcp_port", default_value="-1")
     wait_for_camera_launch_arg = DeclareLaunchArgument("wait_for_camera", default_value="False")
+    objects_file_launch_arg = DeclareLaunchArgument("objects_file", default_value=ensenso.EMPTY_STRING)
+    objects_frame_launch_arg = DeclareLaunchArgument("objects_frame", default_value=ensenso.EMPTY_STRING)
+    visualization_marker_topic_launch_arg = DeclareLaunchArgument("visualization_marker_topic", default_value=ensenso.EMPTY_STRING)
```
Comment (benthie) [inline ensenso_camera2/launch/stereo_node.launch.py:67]:
```
    visualization_marker_topic_launch_arg = DeclareLaunchArgument(
        "visualization_marker_topic", default_value=ensenso.EMPTY_STRING
    )
```

**Review by benthie — CHANGES_REQUESTED** [review]
`stereo_node.launch.py` also needs some reformatting. Not sure if you have black, so I just posted the changes requested by the code-checker before I start the pipeline again ;)

File: ensenso_camera2/launch/stereo_node.launch.py (hunk 1 above)
Comment (Erblinium) [reply ensenso_camera2/launch/stereo_node.launch.py:21]: Can I just suggest and merge it here...?
```suggestion
    visualization_marker_topic = ensenso.fix_empty_string(
        LaunchConfiguration("visualization_marker_topic").perform(context)
    )
```

Comment (Erblinium) [reply ensenso_camera2/launch/stereo_node.launch.py:21]: Nevermind, that was a test. I will just amend the commit again.

File: ensenso_camera/include/ensenso_camera/ros2/node.h [hunk 4]
```diff
@@ -232,3 +236,4 @@ struct NodeHandleWrapper
 #define CREATE_NODE_HANDLE_WRAPPER(name)                                                                               \
   ensenso::ros::NodeHandleWrapper name = ensenso::ros::NodeHandleWrapper(getNodeHandle(), getPrivateNodeHandle());
 #endif
+
```
Comment (benthie) [inline ensenso_camera/include/ensenso_camera/ros2/node.h:239]: The formatter only accepts one trailing newline at the end of the file.

**Comment by benthie** [comment]
Sometimes - we don't know why yet - one or more builds of the pipeline randomly fail. I just re-run the one that failed and it should be ready in a couple of minutes. I will then merge. Thanks again for contributing =)

**Review by benthie — APPROVED** [review]

**Pull request merged**